scripts/machine-launch-script.sh

```diff
@@ -1,83 +1,260 @@
+    # not sure if we need both, but I'm including them for now
+    CONDA_PACKAGE_SPECS+=(
+        sbt \
+        ca-certificates \
+        mosh \
+        gmp \
+        mpfr \
+        mpc \
+        zlib \
+	libzlib \
+        vim \
+        git  \
+        openjdk \
+        gengetopt \
+        libffi \
+        expat \
+        libusb1 \
+        ncurses \
+        cmake \
+        graphviz \
+        expect
+        dtc \
+    )
+
+
+    # Why do we need these?
+    #   "perl(ExtUtils::MakeMaker)" \
+    # diffstat texi2html 
+
+    # verilator was added to conda-forge by timsnyder
+    CONDA_PACKAGE_SPECS+=( verilator==4.034 )
+
+    
+    # python packages
+    # While it is possible to install using pip after creating the
+    # conda environment, pip's dependency resolution can conflict with
+    # conda and create broken environments.  It's best to use the conda
+    # packages so that the environment is consistent
+    CONDA_PACKAGE_SPECS+=( \
+	boto3==1.20.21 \
+        colorama==0.4.3 \
+	argcomplete==1.12.3 \
+	python-graphviz==0.19 \
+	pyparsing==3.0.6 \
+	numpy==1.19.5 \
+	kiwisolver==1.3.1 \
+	matplotlib==3.3.4 \
+	pandas==1.1.5 \
+	awscli==1.22.21 \
+	pytest==6.2.5 \
+	moto==2.2.19 \
+	pyyaml==5.4.1 \
+    )
+
+    if [[ "$CONDA_ENV_NAME" == "base" ]]; then
+	# we default to installing into the base environment against best practice because it
+	# simplifies adding our environment as the default to the system-wide /etc/profile.d/conda.sh
+	CONDA_SUBCOMMAND=install
+	CONDA_ENV_BIN="${CONDA_INSTALL_PREFIX}/bin"
+    else
+	CONDA_SUBCOMMAND=create
+	CONDA_ENV_BIN="${CONDA_INSTALL_PREFIX}/envs/${CONDA_ENV_NAME}/bin"
+    fi
+
+    # to enable use of sudo and avoid modifying 'secure_path' in /etc/sudoers, we specify the full path to pip
+    CONDA_PIP_EXE="${CONDA_ENV_BIN}/pip"
+
+    # to enable use of sudo and avoid modifying 'secure_path' in /etc/sudoers, we specify the full path to conda
+    $SUDO "${CONDA_EXE}" "$CONDA_SUBCOMMAND" $DRY_RUN_OPTION -n "$CONDA_ENV_NAME" -y "${CONDA_PACKAGE_SPECS[@]}"
+
+
+    # Install python packages using pip that are not available from conda
+    #
+    # Installing things with pip is possible.  However, to get
+    # the most complete solution to all dependencies, you should
+    # prefer creating the environment with a single invocation of
+    # conda
+    PIP_PKGS=(fab-classic==1.19.1)
+    if [[ -n "$PIP_PKGS[*]" ]]; then
+	"${DRY_RUN_ECHO[@]}" $SUDO "${CONDA_PIP_EXE}" install fab-classic==1.19.1
+    fi
+
+    # setup argcomplete
+    "${DRY_RUN_ECHO[@]}" $SUDO "${CONDA_ENV_BIN}/activate-global-python-argcomplete"
+
+    conda_init_extra_args=()
+    if [[ -n "$SUDO"  || "$USER" == root ]]; then
+	# if we're installing into a root-owned directory using sudo, or we're already root
+	# initialize conda in the system-wide rcfiles
+	conda_init_extra_args=(--no-user --system)
+    fi
+    $SUDO "${CONDA_EXE}" init $DRY_RUN_OPTION "${conda_init_extra_args[@]}" bash
```

Comment:
This should move up to be inside the conditioning of doing the conda install and not happen unless we're actually installing conda (as opposed to reusing an existing install in `--prefix`)